### PR TITLE
test(quantic): E2E tests for for QuanticRecentResultsList component

### PR DESCRIPTION
**Pull request by aelfaleh**
image

### CODE REVIEW

File: packages/quantic/cypress/integration/recent-results-list/recent-results-list-expectations.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      selector
+        .results()
+        .should('have.length', length)
+        .logDetail(`should display ${length} results in the list`);
+    },
+    lastResultContains: (result: string) => {
+      selector
+        .lastResult()
+        .contains(result)
+        .logDetail(`The last result added should contain "${result}"`);
```
Comment (lbergeron) [inline packages/quantic/cypress/integration/recent-results-list/recent-results-list-expectations.ts:54]:
```suggestion
        .logDetail(`The last added result should contain "${result}"`);
```

File: packages/quantic/cypress/integration/recent-results-list/recent-results-list-expectations.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        .label()
+        .contains(label)
+        .logDetail(`The label should contain "${label}"`);
+    },
+    hrefContainsAtResult: (value: string) => {
+      selector
+        .result(value)
+        .should('have.attr', 'href')
+        .and('include', urlResult)
+        .then((href) => expect(href).to.equal(urlResult))
```
Comment (lbergeron) [inline packages/quantic/cypress/integration/recent-results-list/recent-results-list-expectations.ts:67]: These two lines seem to validate the same thing.

The `.then((href) => expect...)` seems to be stricter, so can we keep only this one and remove the `.and('include', urlResult)`?

File: packages/quantic/cypress/integration/recent-results-list/recent-results-list-expectations.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    logDocumentOpen: (resultTitle: string) => {
+      cy.wait(InterceptAliases.UA.DocumentOpen)
+        .then((interception) => {
+          const analyticsBody = interception.request.body;
+          expect(analyticsBody).to.have.property('actionCause', 'documentOpen');
+          expect(analyticsBody).to.have.property('documentTitle', resultTitle);
+          expect(analyticsBody).to.have.property('documentUri', urlResult);
+        })
+        .logDetail('should log the "documentOpen" UA event');
+    },
```
Comment (lbergeron) [inline packages/quantic/cypress/integration/recent-results-list/recent-results-list-expectations.ts:91]: We have [this expectation](https://github.com/coveo/ui-kit/blob/master/packages/quantic/cypress/integration/common-expectations.ts) that could be reused here.

Your code would look like this:

```suggestion
    logDocumentOpen: (resultTitle: string) =>
      logUaEvent(
        InterceptAliases.UA.DocumentOpen,
        'documentOpen',
        {
          actionCause: 'documentOpen',
          documentTitle: resultTitle,
          documentUri, urlResult
        }
      ),
```

File: packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { api, LightningElement } from "lwc";
+
+export default class ActionSetLocalStorage extends LightningElement {
+    @api engineId;
+
+    localStorageKey =
+  'quantic-recent-results-list-engine_quantic-recent-results';
```
Comment (lbergeron) [inline packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.js:7]: Indent is weird for this file.

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticRecentResultsList/exampleQuanticRecentResultsList.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            attribute: 'isCollapsed',
+            label: 'Is collapsed',
+            description: 'Whether the component is collapsed.',
+            defaultValue: false
+        },
+        {
+            attribute: 'target',
+            label: 'Target',
+            description: ' Where to display the linked URLs, as the name for a browsing context.',
```
Comment (lbergeron) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticRecentResultsList/exampleQuanticRecentResultsList.js:32]:
```suggestion
            description: 'Where to display the linked URLs, as the name for a browsing context.',
```

**Review by lbergeron — APPROVED** [review]

File: packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.js (hunk 4 above)
Comment (nathanlb) [reply packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.js:7]: That's a looooooooooog key! haha

File: packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        uri: 'https://github.com/coveo/ui-kit/',
+        uniqueId: '2',
+        clickUri: 'https://github.com/coveo/ui-kit/',
+        raw: {
+            urihash: '2',
+        },
+        },
+    ];
+
+    setItem() {
```
Comment (nathanlb) [inline packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.js:29]: given we're setting two items maybe we `setItems` would be more accurate.

File: packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+      onclick={setItem}
+      class="configurator__set-local-storage">
+    </lightning-button>
+    <lightning-button
+      label="Clear localStorage"
+      title="Clear localStorage"
+      onclick={clear}
+      class="configurator__clear-local-storage">
+    </lightning-button>
+  </template>
```
Comment (nathanlb) [inline packages/quantic/force-app/examples/main/lwc/actionSetLocalStorage/actionSetLocalStorage.html:14]: slight indentation issues in this file

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticRecentResultsList/exampleQuanticRecentResultsList.html [hunk 8]
```diff
@@ -0,0 +1,22 @@
+<template>
```
Comment (nathanlb) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticRecentResultsList/exampleQuanticRecentResultsList.html:1]: this file has 4-space indentation

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticRecentResultsList/exampleQuanticRecentResultsList.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import {api, LightningElement, track} from 'lwc';
```
Comment (nathanlb) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticRecentResultsList/exampleQuanticRecentResultsList.js:1]: also a bit too much indentation here

File: packages/quantic/cypress/page-objects/actions/action-set-local-storage.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+export const setRecentResultsListLocalStorage = () =>
+  cy.get('c-action-set-local-storage .configurator__set-local-storage').click();
```
Comment (nathanlb) [inline packages/quantic/cypress/page-objects/actions/action-set-local-storage.ts:2]: should we log an action here?

File: packages/quantic/cypress/page-objects/actions/action-set-local-storage.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+export const clearLocalStorage = () =>
+  cy
```
Comment (nathanlb) [inline packages/quantic/cypress/page-objects/actions/action-set-local-storage.ts:5]: same. log action?

File: packages/quantic/cypress/integration/recent-results-list/recent-results-list.cypress.ts [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+    Expect.displayPlaceholder(true);
+    Expect.displayResults(false);
+  });
+
+  describe('with default options', () => {
+    beforeEach(() => {
+      clearLocalStorage();
+    });
+    it('should work as expected', () => {
```
Comment (nathanlb) [inline packages/quantic/cypress/integration/recent-results-list/recent-results-list.cypress.ts:50]: little newline here might look a little less cluttered.

**Review by nathanlb — APPROVED** [review]
This is really nice work! 🎉

**Pull request merged**